store/README.md

```diff
@@ -0,0 +1,89 @@
+# API Rationale
```

Comment:
Here's an updated doc with some explanation of how the current API behaves. Might help to a more high-level feeling of how it looks now.

Reply:
LOVE LOVE LOVE